### DISCUSSION TITLE
Query regarding the Span parameters?

### DISCUSSION THREAD

**Discussion by Palzar-Tamang** [Telecominfraproject/oopt-gnpy#475] [Q&A] [answered]
Hi,
Had a doubt about what does this "target_extended_gain" mean. 
In some network planning scenarios, you might need an amplifier to provide a gain that is outside of its typical range. This is where "targeted extended gain" comes into play. It allows you to specify a desired gain value that goes beyond the usual limits of the amplifier.
So now, if you have an amplifier with a typical gain range of -5 dBm to +10 dBm, you might use the "targeted extended gain" feature to request a gain of +15 dBm, which is beyond the amplifier's typical maximum gain. 
Is this what it meant by "target_extended_gain"?

Thankyou
Palzar

**Comment by Palzar-Tamang** [comment]
Could someone help me with this please!!!

**Reply by EstherLerouzic** [reply, marked as answer]
Hello @Palzar-Tamang
I am a bit confused by your question: you give an example with a gain range in dBm but gain is in dB and always positive. do you mean typical power range  ?
The target_extended_gain does what you have identified:
when no amplifier type-variety is set in an amplifier node, the program computes the target power and search for the best amplifier within the library that satisfies this power provided that the gain is kept within  the max flat gain + target_extended_gain, and total power is below max power.
Now if the type-variety is already set in the node, the program just uses the operational values that have been set. The program warns the user if the gain exceeds the max flat-gain +  target_extended_gain.

Hope this helps
Esther

**Reply by Palzar-Tamang** [reply]
Sorry for the confusion I meant to say power range -5dBm to 10dBm .
Thank you for the answer.

**Reply by Palzar-Tamang** [reply]
Could you also tell what is meant by delta power range db and the function of power mode?

**Reply by EstherLerouzic** [reply]
Hello,

You may find the answer to your question in this discussion: https://github.com/Telecominfraproject/oopt-gnpy/discussions/472#discussioncomment-6822757. in short:
"The optimum input power per span is not the same for all spans (depends on length and loss), so delta_power_range_db enables a range around power_dbm to optimize the input power per span. This deviation around power_dbm is recorded into delta_p in each amplifier instance in the topology."

hope this helps,

Esther

**Reply by Palzar-Tamang** [reply]
Thanks for answer
